pkg/asset/installconfig/aws/validation.go

```diff
@@ -166,6 +184,167 @@ func validateAMI(ctx context.Context, config *types.InstallConfig) field.ErrorLi
+		return nil
+	}
+	var eipAllocations []awstypes.EIPAllocation
+	if config.Platform.AWS != nil && config.Platform.AWS.EIPAllocations != nil && len(config.Platform.AWS.EIPAllocations.IngressNetworkLoadBalancer) == 0 {
+		eipAllocations = config.Platform.AWS.EIPAllocations.IngressNetworkLoadBalancer
+	}
+	var allocationIDs []string
+	for _, allocationID := range eipAllocations {
+		allocationIDs = append(allocationIDs, string(allocationID))
+	}
+
+	// Validate EIP allocation IDs
+	if err := validateEIPAllocationsList(allocationIDs); err != nil {
+		fmt.Printf("Validation error: %v\n", err)
+	} else {
+		fmt.Println("All allocation IDs are valid.")
+	}
+
+	// Check if eipAllocations exist
+	for _, allocationID := range allocationIDs {
+		_, err := eipAllocationExists(client, allocationID)
+		if err != nil {
+			err = fmt.Errorf("eipAllocations does not exists due to error %s:", err)
+			return append(allErrs, field.InternalError(fldPath, err))
+		}
+	}
+
+	// Get all associated EIPs in the list
+	associatedEIPs, err := getAssociatedEIPs(client, allocationIDs)
+	if err != nil {
+		err = fmt.Errorf("could not get check if eips are associated due to error %s:", err)
+		return append(allErrs, field.InternalError(fldPath, err))
+	}
+	fmt.Printf("Asociated eips %v", associatedEIPs)
+
+	// Print error if EIPs are associated
+	if len(associatedEIPs) > 0 && config.Platform.AWS != nil && config.Platform.AWS.EIPAllocations != nil && len(config.Platform.AWS.EIPAllocations.IngressNetworkLoadBalancer) == 0 {
+		return append(allErrs, field.Invalid(fldPath, associatedEIPs, fmt.Errorf("eipAllocations %s can't be used as %s are already associated", config.Platform.AWS.EIPAllocations.IngressNetworkLoadBalancer, associatedEIPs).Error()))
+	}
+
+	// BYO Subnets
+	if len(config.Platform.AWS.Subnets) != 0 {
+		if err = compareBYOSubnetsWithEIPAllocations(config); err != nil {
+			return append(allErrs, field.Invalid(fldPath, nil, compareBYOSubnetsWithEIPAllocations(config).Error()))
+		}
+	} else { // Installer Created Subnets
+		allzones, err := meta.AvailabilityZones(ctx)
+		if err != nil {
+			return append(allErrs, field.InternalError(fldPath, err))
+		}
+		if err = compareInstallerCreatedSubnetsWithEIPAllocations(config, allzones); err != nil {
+			return append(allErrs, field.Invalid(fldPath, nil, compareInstallerCreatedSubnetsWithEIPAllocations(config, allzones).Error()))
+		}
+	}
+	return nil
+}
+
+func compareBYOSubnetsWithEIPAllocations(config *types.InstallConfig) error {
+	if len(config.Platform.AWS.Subnets) != len(config.Platform.AWS.EIPAllocations.IngressNetworkLoadBalancer) {
+		return fmt.Errorf("subnet count %s != eipAllocations count %s", config.Platform.AWS.Subnets, config.Platform.AWS.EIPAllocations.IngressNetworkLoadBalancer)
+	}
+	return nil
+}
+
+func compareInstallerCreatedSubnetsWithEIPAllocations(config *types.InstallConfig, allzones []string) error {
```

Comment:
We don't need this function. See https://github.com/openshift/installer/pull/8204/commits/af6958b238d8c7a28e14f57a98c593217eb501a1#r1849022269.

Reply:
same answer as above https://github.com/openshift/installer/pull/8204#discussion_r1849323129

Reply:
Please try again to make this work without duplicated checks.

Reply:
which duplicate checks ?

Reply:
Details here: https://github.com/openshift/installer/pull/8204/files#r1854624041